File: app/com/zapodot/metrics/MetricsPlugin.java
package com.zapodot.metrics;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import play.Play;
import play.Plugin;

/**
 * @author zapodot at gmail dot com
 */
public class MetricsPlugin extends Plugin{

    private static Logger logger = LoggerFactory.getLogger(MetricsPlugin.class);

    @Override
    public void onStart() {
        logger.info("Starting MetricsPlugin");
        if(MetricsSetup.class.isAssignableFrom(Play.application().getWrappedApplication().global().getClass())) {
            logger.info("MetricsSetup has been enabled. Will time and meter all requests, and count all errors");
        }
    }

    @Override
    public void onStop() {
        logger.info("Stopping MetricsPlugin");
    }
}

File: app/org/zapodot/metrics/MetricsPlugin.scala
import play.api.{Logger, Plugin, Application}

package org.zapodot.metrics {

import play.api.mvc.{WithFilters, Result, RequestHeader, Filter}
import com.codahale.metrics.Timer.Context
import play.api.GlobalSettings

/**
 * Plugin class that plugs Metrics into a Play Framework application.
 * @param app
 */
class MetricsPlugin(app: Application) extends Plugin {

  val logger = Logger

  override def onStart() {
    logger.info("Starting MetricsPlugin")
    if(app.global == MetricsSetup) {
      logger.info("The MetricsSetup Global setting has been enabled. All requests will be timed and all controller errors counted")
    }
  }
}

/**
 * Global object that enables error counter and Timer
 */
object MetricsSetup extends WithFilters(TimerFilter) with GlobalSettings {
  override def onError(request: RequestHeader, ex: Throwable): Result = {
    Metrics.countThrowable(request, ex);
    super.onError(request, ex)
  }
}

/**
 * Filter to be used for timing requests
 */
object TimerFilter extends Filter {
  def apply(next: (RequestHeader) => Result)(requestHeader: RequestHeader): Result = {
    val timeContext: Context = Metrics.timerForRequest(requestHeader).time()
    val result = next(requestHeader)
    timeContext.stop();
    result
  }
}



}